### PR TITLE
Async responses

### PR DESCRIPTION
**Pull request by jcaw**
Have grammars send async responses, fix from a while ago. Need this to get Dragonfly working.

### CODE REVIEW

**Comment by lunixbochs** [comment]
With the public Talon release happening, this is now a higher priority - can you schedule some time to work with me to test / merge this soon with Talon?

**Comment by jcaw** [comment]
Sure, sounds good. Let me know on Slack. 

My Dragonfly branch has a client & test suite. It's been a while but I _think_ everything passes. https://github.com/jcaw/dragonfly/tree/draconity-backend-2/dragonfly/test/test_draconity

File: src/transport/client.h [hunk 1]
```diff
@@ -66,10 +71,14 @@ class UvClient : public UvClientBase {
         } else {
             reply = handle_message_callback(this->id, received_header->tid, msg);
         }
-        uint32_t reply_length;
-        uint8_t *reply_data = bson_destroy_with_steal(reply, true, &reply_length);
-        writeMessage(received_header->tid, reply_data, reply_length);
-        bson_free(reply_data);
+        // HACK: Some messages won't return a reply immediately. If a message
+        //   returns null, it's making a pinky promise to reply later.
```
Comment (jcaw) [inline src/transport/client.h:75]: The client will probably need to adapt to this change

File: src/server.cpp [hunk 2]
```diff
@@ -451,7 +451,8 @@ static bson_t *handle_message(uint64_t client_id, uint32_t tid, const std::vecto
         draconity->mimic_lock.lock();
         draconity->mimic_queue.push({client_id, tid});
         draconity->mimic_lock.unlock();
-        resp = success_msg();
+        // Response will be sent once mimic completes.
```
Comment (jcaw) [inline src/server.cpp:454]: Mimic will need to be adapted to this message scheme in the client

File: src/server.cpp [hunk 3]
```diff
@@ -567,10 +569,10 @@ void draconity_mimic_done(int key, dsx_mimic *mimic) {
         draconity->mimic_lock.unlock();
         uint64_t client_id = mimic_info.first;
         uint32_t tid = mimic_info.second;
-        draconity_publish_one("mimic",
-                              BCON_NEW("cmd", "mimic.done",
-                                       "tid", BCON_INT32(tid)),
-                              client_id);
+        draconity_send("mimic",
+                       success_msg(),
```
Comment (jcaw) [inline src/server.cpp:573]: Before there were two mimic messages, one for scheduled, one for completion. This should be the only mimic-related message now.

**Pull request merged**